mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -742,18 +742,34 @@ fn dex_fee_rate(base: &str, rel: &str) -> MmNumber {
     }
 }
 
-pub fn dex_fee_amount(base: &str, rel: &str, trade_amount: &MmNumber, min_tx_amount: &MmNumber) -> MmNumber {
+pub fn dex_fee_amount(base: &str, rel: &str, trade_amount: &MmNumber, min_tx_amount: &MmNumber) -> DexFee {
     let rate = dex_fee_rate(base, rel);
-    let fee_amount = trade_amount * &rate;
-    if &fee_amount < min_tx_amount {
-        min_tx_amount.clone()
-    } else {
-        fee_amount
+    let fee = trade_amount * &rate;
+
+    if &fee < min_tx_amount {
+        return DexFee::Standard(min_tx_amount.clone());
+    }
+
+    if base == "KMD" {
+        // Drop the fee by 25%, which will be burned during the taker fee payment.
+        //
+        // This cut will be dropped before return if the final amount is less than
+        // the minimum transaction amount.
+        let new_fee = &fee * &MmNumber::from("0.75");
+        let burn_amount = &fee - &new_fee;
+
+        if &new_fee >= min_tx_amount {
+            return DexFee::with_burn(new_fee, burn_amount);
+        } else {
+            return DexFee::Standard(min_tx_amount.clone());
+        }
```

Comment:
We did this intentionally (only burn when fee is equal or greater than min tx after the 25% drop).

Reply:
Could you please provide more info on motivation on this? Also, the code reduces the fee amount to be paid anyway in the edge case, which I consider as bug.

Reply:
I missed the point. If there is no burn, fee shouldn't be reduced. Good catch!

Reply:
> Also, the code reduces the fee amount to be paid anyway in the edge case, which I consider as bug.

https://github.com/KomodoPlatform/komodo-defi-framework/pull/2006/commits/7e05366dd2f75fbe31ce3cd1be9dd41172960a1b this should fix that.

Reply:
> cc @shamardy, would like to have your opinion on this as well 🙂

I think we discussed this on DMs (KMD burn channel), my opinion was that if the total fee is between  `min_tx_amount` and `1.25 * min_tx_amount` the excess to `min_tx_amount` can be burned.

P.S. There can be a problem in the future when we implement conversion from other coins to KMD to burn this amount in KMD, in this case the burn amount has to be at least the `min_tx_amount` of the coin so that it can be sent in an output to be converted if we decided to do this burning in a decentralized manner.

Reply:
I have no opposing-feelings to "burn up to 25% (even if the burn amount is less than 25%)", I guess that's why you both are suggesting, should I update the calculations that way?

Reply:
> should I update the calculations that way?

Yes please :)

Reply:
Updated with https://github.com/KomodoPlatform/komodo-defi-framework/pull/2006/commits/d817f00d957c9bbc698da2fb2b78c617cbb09932